draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,528 +1,448 @@
+   distinct from the inputs to CC.  The latter calculates a sending rate
+   or window from network measurements, and it takes the data to send as
+   input, sometimes along with application requirements such as upper/
+   lower rate bounds, periods of quiescence, or a priority.
+
+   It is not clear that the ACK signals feeding into a congestion
+   control algorithm are useful to FEC in their raw form, and vice versa
+   - information about repaired blocks may be quite irrelevant to a CC
+   algorithm.  However, there can be meaningful other interactions
+   (indicated by the horizontal double arrow) between the two entities,
+   usually as a result of their operation rather than by relaying their
+   own raw inputs.  For example, the network measurements carried out by
+   CC can yield a longer-term statistical measure such as a loss ratio
+   which is useful input for a FEC coding scheme.  Similarly, unequal
+   error protection using fountain codes can be used to assign different
+   priorities to blocks of data, and these priorities can be honored by
+   a CC mechanism.
+
+3.  Scope
+
+   This section describes the scope of the document.
+
+3.1.  Type of application
+
+   The document focuses on reliable transmissions.
+
+3.2.  End-to-end
+
+   The document focuses on end-to-end coding, i.e. cases where coding is
+   added at the server and client end points.  The discussions should
+   then consider fairness with non-coding solutions.
+
+
+
+
+Kuhn, et al.              Expires July 22, 2020                 [Page 4]
+
+Internet-Draft            Coding and congestion             January 2020
+
+
+3.3.  Objective of the document
+
+   Section 2 presents that FEC and CC can be seen as two separate
+   channels.  In practice, implementations may mix the signals that are
+   exchanged on these channels.  This document discusses how FEC and CC
+   relate.  [NK: We may want to later add that we provide
+   recommendations - but for the moment, we may just list the server
+   side coding solutions in the layering approach without providing
+   recommendations for each case]
+
+4.  FEC above the transport
+
+   Figure 3 illustrates the FEC above the transport case.
+
+                 |                  --------                  ^
+                 | data             |      | data,            | sending
+                 |                  |      | requirements     | rate (or
+                 v                  |      v                  | window)
+  +-----------------------------+   |    +--------------------------+
+  |             FEC             |        | Transport (including CC) |
+  +-----------------------------+   |    +--------------------------+
+    ^                   |           |              ^
+    | information       | coded     |              | network
+    | about repaired    | data      |              | measurements
+    | ratio             -------------              |
+
+                     Figure 3: FEC above the transport
```

Comment:
> I have put the boxes side by side in figure 2 above to avoid discussing layering and just discuss the separate functionality. 
I think that the current Figure 2 is an example where FEC is on top of the CC, but interactions between the FEC and CC are possible. I have proposed an updated Figure in the commit.

Reply:
I'm not sure I see the update... anyway, sure, interactions are possible!  If you want to add more arrows between them, fine! But still I would layer the boxes such that FEC is above transport.